libminifi/include/utils/file/FileSystem.h

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <memory>
+#include "utils/OptionalUtils.h"
+#include "utils/EncryptionProvider.h"
+#include "core/logging/LoggerConfiguration.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+namespace file {
+
+class FileSystem {
+ public:
+  explicit FileSystem(bool should_encrypt = false, utils::optional<utils::crypto::EncryptionProvider> encryptor = {});
+
+  FileSystem(const FileSystem&) = delete;
+  FileSystem(FileSystem&&) = delete;
+  FileSystem& operator=(const FileSystem&) = delete;
+  FileSystem& operator=(FileSystem&&) = delete;
+
+  utils::optional<std::string> read(const std::string& file_name);
+
+  bool write(const std::string& file_name, const std::string& file_content);
+
+ private:
+  bool should_encrypt_on_write_;
+  utils::optional<utils::crypto::EncryptionProvider> encryptor_;
+  std::shared_ptr<logging::Logger> logger_{logging::LoggerFactory<FileSystem>::getLogger()};
+};
```

Comment:
The `FileSystem` name and the utils::file namespace is most likely too general here, because a real filesystem can not be modeled by a boolean and optional EncryptionProvider structure.

Reply:
agree, do you have a specific name in mind?

Reply:
I didn't yet fully review the PR, so the usage is not clear to me at the moment.

Reply:
Maybe `EncryptionAwareFileAccessor`, `ConfigFileIo` or some combination of these words. Or refactor to use streams from either `minifi::io::InputStream`/`minifi::io::OutputStream` or `<iostream>` and insert an encryption/decryption stream in the chain when needed.

Reply:
I specifically wanted it not to be aware of encryption, `ConfigFileIO` seemed a little too specific for me, but for the current purposes this seems to be our best bet

Reply:
the problem with this, now that I think about it, is the question of "who owns the config file path", because if it is a `ConfigFileIO` it should own the path and read/write should not get a path parameter, but then again currently `FlowConfiguration` owns the config file path, so I think that would be a bigger change (which may not be beneficial at all)

Reply:
how about `FileIO`?